### PR TITLE
#106 Fixed is_in_stock for configurable products accordance by childs…

### PR DESCRIPTION
**Pull request by dimasch**
Fix for #106  \+ @afirlejczyk @pkarw

### CODE REVIEW

File: src/module-vsbridge-indexer-catalog/Model/Indexer/DataProvider/Product/ConfigurableData.php [hunk 1]
```diff
@@ -289,6 +289,11 @@ private function prepareConfigurableProduct(array $productDTO)
             $productDTO['stock']['stock_status'] = 0;
         }
 
+        if (!$areChildInStock) {
+            $productDTO['stock']['is_in_stock'] = false;
+            $productDTO['visibility'] = 1;
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-catalog/Model/Indexer/DataProvider/Product/ConfigurableData.php]: I wouldn't change the visibility, some users want to display out of stock product in catalog (product are loaded base on visibility from ES).

Please make a change, and I will merge PR.

Thanks!

Comment (afirlejczyk) [reply src/module-vsbridge-indexer-catalog/Model/Indexer/DataProvider/Product/ConfigurableData.php]: I will make a little different solution for this. There was a similar problem in magento1-vsbridge-indexer. https://github.com/DivanteLtd/magento1-vsbridge-indexer/pull/24/files

Comment (afirlejczyk) [reply src/module-vsbridge-indexer-catalog/Model/Indexer/DataProvider/Product/ConfigurableData.php]: I change a logic a little and made new PR
https://github.com/DivanteLtd/magento2-vsbridge-indexer/pull/109/files

Comment (dimasch) [reply src/module-vsbridge-indexer-catalog/Model/Indexer/DataProvider/Product/ConfigurableData.php]: Thanks.

**Pull request closed without merging**